iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/GeneratorAdderImpl.java

```diff
@@ -109,19 +117,32 @@ public GeneratorAdder setRatedS(double ratedS) {
     public GeneratorImpl add() {
         String id = checkAndGetUniqueId();
         TerminalExt terminal = checkAndGetTerminal();
+        boolean validateRegulatingTerminal = true;
+        if (useLocalRegulation) {
+            regulatingTerminal = terminal;
+            validateRegulatingTerminal = false;
```

Comment:
I don't see the use of this parameter in validation methods? You can check the terminal even if it is local

Reply:
Validation of regulating terminal checks that the terminal is not null and its network is the same as the object being added. 

`regulatingTerminal != null && regulatingTerminal.getVoltageLevel().getNetwork() == network;`

The network for the terminal is obtained from its voltage level and terminal voltage level is not set until the terminal is attached with the method (`voltageLevel.attach(terminal, false)`). The terminal is attached after the validation is performed and during the validation there is not voltageLevel associated to the terminal.

```
boolean validateRegulatingTerminal = true;
if (useLocalRegulation) {
            regulatingTerminal = terminal;
            validateRegulatingTerminal = false;
}

ValidationUtil.checkVoltageControlTargetQ(this, regulatingTerminal, targetV, targetQ, voltageRegulatorOn,
   getNetwork(), validateRegulatingTerminal);

GeneratorImpl generator
   = new GeneratorImpl(getNetwork().getRef(),
       id, getName(), isFictitious(), energySource,
       minP, maxP,
       voltageRegulatorOn, regulatingTerminal,
       targetP, targetQ, targetV,
       ratedS);
generator.addTerminal(terminal);
voltageLevel.attach(terminal, false);

```

One option is to do the attach before the validation but we do not want to move the order of validation and terminal attachment (if we attach first and then validation fails, we should unroll the changes).
We prefer do not check the regulating terminal if it is local. So for local regulation (useLocalRegulation) we assume the terminal will be ok since it will be the one of the equipment and it is set automatically by the adder method.

Reply:
Ah okay, I did not get this! Indeed, if the terminal is not yet attached, it is not considered part of the network...